### ISSUE TITLE
Rule Proposal: prefer-static-string-properties

### ISSUE THREAD

**Issue by undsoft** [angular-eslint/angular-eslint#1172]
Examples of incorrect code:
```
<div [myInput]="'myString'"></div>
```

Examples of correct code:
```
<div myInput="myString"></div>
```

**Comment by ezzabuzaid** [comment]
The good thing about property binding is that if you removed the binding input or changed its name, Angular will show you an error. for instance, `Can't bind to 'disableRippless' since it isn't a known property of 'mat-checkbox'.`

**Comment by undsoft** [comment]
True, well then the rule can be `property-binding-style` or something like that, supporting both options.

**Comment by JamesHenry** [comment]
I discussed this one with Angular folks today, including Alex Rickabaugh, and he agreed with this proposal as matching the up to date recommendations from them in 2024.
 
I would be happy to accept PRs for this one any time.

For the name, I think `prefer-static-string-properties` might be better, because (A) we will only be checking for this case where a string literal is being bound to (B) that is ultimately the difference:

```html
<div [myInput]="'myString'"></div>
```

^ For legacy reasons (according to Alex) this always creates an element binding even though the value will never change.

```html
<div myInput="myString"></div>
```

^ Whereas this creates a static property.

Relevant compiler output instructions shown here:

https://jeanmeche.github.io/angular-compiler-output/?template=PGRpdiBbbXlhdHRyXT0iJ3Rlc3QnIj48L8SBdj4KxIDEgiDEhsSIxIoyxI3EkMSSxJU%3D
